src/server/codec.rs

```diff
@@ -368,6 +368,7 @@ pub struct Spawn {
     /// If None, create a new window for this new tab
     pub window_id: Option<WindowId>,
     pub command: Option<CommandBuilder>,
+    pub command_dir: Option<String>,
```

Comment:
github won't let me comment on lines that aren't in the diff, so I'm picking this one :-p

I think you'll also want to add the dir to `WindowAndTabEntry` above so that we can record it when the domain is attached, and to `GetTabRenderChangesResponse` so that we can notice and update when the remote cwd is updated.

(I think you'll want to look for any place that we're dealing with the `title` and add similar processing for the cwd)